### PR TITLE
Standardize CI setup

### LINKED ISSUE #177: Standardize CI infrastructure
- [ ] put owncloud/coding-standard in ``vendor-bin``
- [ ] rename ``tests/Unit`` to ``tests/unit`` (case of ``Unit`` is non-standard) and review related scripts
- [ ] add ``test-acceptance-*`` to ``Makefile`` and use it from drone
- [ ] review the CI infrastructure and standardize anything else

Part of https://github.com/owncloud/QA/issues/582

### PR DESCRIPTION
**Pull request by skshetry**
Standardize CI setup for `owncloud/oauth2`.

Fixes #177 .

- [x] Add `NEED_CORE`.
- [x] Indent `drone.yml` commands.
- [x] Use makefile in drone instead of `run.sh`.
- [x] Run unittest with dbg only when `COVERAGE: true`.
- [x] Add coding standard
- [x] Add few folders to `.gitignore`.
- [x] Standardize makefile
- [x] Add `bamarni` and setup `vendor-bin`.
- [x] PHAN/PHPSTAN
- [x] Use bootstrap of core

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -59,35 +75,52 @@ pipeline:
     image: owncloudci/php:${PHP_VERSION}
     pull: true
     commands:
```
Comment (phil-davis) [inline .drone.yml]: this is missing an environment section like:
```
    environment:
      - COVERAGE=${COVERAGE}
```

File: .drone.yml [hunk 2]
```diff
@@ -59,35 +75,52 @@ pipeline:
...
   codecov:
     image: plugins/codecov:2
     secrets: [codecov_token]
     pull: true
     files:
-    - '*.xml'
+      - '*.xml'
     paths:
-    - tests/output
+      - tests/output
```
Comment (phil-davis) [inline .drone.yml]: make it the same as all the other ``codecov`` sections:
```
    paths:
      - tests/output/*.xml
```
and no ```files`` section.

File: .gitignore [hunk 3]
```diff
@@ -1,5 +1,12 @@
 /build/
-clover.xml
+
 .php_cs.cache
```
Comment (phil-davis) [inline .gitignore]: The ``l10n`` line is also handy to have, like other apps:
```
/l10n/.transifexrc
```

**Comment by skshetry** [comment]
@phil-davis  please review again. Fixed that.

**Pull request merged**